### PR TITLE
Ddan 118 execute filesoption

### PR DESCRIPTION
**Pull request by HAN-oQo**
### PURPOSE OF PR
* Simple summary of this PR

Add "files" option to yaml file which is used in "exs execute -f <yaml>".
The file option contains a list of file names to be uploaded before proceeding the experiments in the task manager.

### What type of PR is it?
* fix / feature / improvement /  hot-fix / refactoring

feature

### JIRA ISSUE
* https://ddangcoonfire.atlassian.net/browse/DDAN-118

### HAVE YOU CHECKED ...
* [x] ADD JIRA ID ON PR TITLE (ex. [DDAN-1234] PR TITLE)
* [x] pre-commit
* [x] Branch Format ( type/issue-NUM ) (ex. refactoring/DDAN-1234 )

### Explanation
( Adding screenshot would be great help )
image
image
image
image

Before executing experiments in task manager, we need to upload required files. So we add "files" option in yaml file.
It makes submitter to upload all files in the file list before execute experiments.

### How to Test (Optional)
1) write the python file somewhere you want to. (recommend to make in different directory of exs)
2) add files option to yaml file, and give list with file_path as input
3) execute with the yaml file
4) the file should be downloaded to the exs directory.

### CODE REVIEW

File: experiment_scheduler/master/master.py [hunk 1]
```diff
@@ -392,6 +395,14 @@ def execute_task(self, task_manager_address, task_id):
             task.commit()
         return response
 
+    @start_end_logger
+    def upload_file(self, request_iterator, context):
+        for request in request_iterator:
```
Comment (sinunu) [inline experiment_scheduler/master/master.py:400]: Please send data to only task_manager which execute task

File: experiment_scheduler/submitter/execute.py [hunk 2]
```diff
@@ -41,7 +45,7 @@ def parse_input_file(parsed_yaml):
 
 
 def main():
-    """
+    """f
```
Comment (sinunu) [inline experiment_scheduler/submitter/execute.py:48]:
```suggestion
    """
```

File: experiment_scheduler/submitter/execute.py [hunk 3]
```diff
@@ -56,6 +60,27 @@ def main():
     stub = master_pb2_grpc.MasterStub(channel)
 
     request = parse_input_file(parsed_yaml)
+    files = [task["files"] if "files" in task else [] for task in parsed_yaml["tasks"]]
```
Comment (sinunu) [inline experiment_scheduler/submitter/execute.py:63]: Please separate this part as a function.

File: experiment_scheduler/task_manager/task_manager_server.py [hunk 4]
```diff
@@ -217,6 +217,13 @@ def get_task_log(self, request, context):
                 log_file=None, error_message=bytes(error_message, "utf-8")
             )
 
+    @start_end_logger
+    def upload_file(self, request_iterator, context):
```
Comment (sinunu) [inline experiment_scheduler/task_manager/task_manager_server.py:221]: Please add a code to remove existing file if this function is called once.

**Review by sinunu — COMMENTED** [review]
Thanks for good feature. I left some comments. please take a look.

File: experiment_scheduler/master/master.py [hunk 5]
```diff
@@ -392,6 +395,14 @@ def execute_task(self, task_manager_address, task_id):
...
+            self.process_monitor.upload_file(request)
+        return MasterResponse(
+            experiment_id="0", response=MasterResponse.ResponseStatus.SUCCESS
+        )
```
Comment (Sangmin0728) [inline experiment_scheduler/master/master.py:404]: I think 'experiment_id' is unuseful.
So, I suggest that this method return new type of message, like 
```suggestion
        return UploadResponse(
            file_name = "(ex.uploaded file name)", response=MasterResponse.ResponseStatus.SUCCESS
        )
```

**Review by Sangmin0728 — APPROVED** [review]
Thanks for developing new feature.
I just suggest only one part! Please check the comment.

**Comment by HAN-0Q0** [comment]
230524
1. bug in "abnormal_exited_task()"
2. unique file name setting

**Pull request merged**